x/concentrated-liquidity/msg_server.go

```diff
@@ -149,9 +153,13 @@ func (server msgServer) CollectIncentives(goCtx context.Context, msg *types.MsgC
 		return nil, err
 	}
 
-	collectedIncentives, err := server.keeper.collectIncentives(ctx, sender, msg.PositionId)
-	if err != nil {
-		return nil, err
+	totalCollectedIncentives := sdk.NewCoins()
+	for _, positionId := range msg.PositionIds {
+		collectedIncentives, err := server.keeper.collectIncentives(ctx, sender, positionId)
+		if err != nil {
+			return nil, err
+		}
+		totalCollectedIncentives = totalCollectedIncentives.Add(collectedIncentives...)
```

Comment:
ditto for emitting event per position id

Reply:
Same as above, added here https://github.com/osmosis-labs/osmosis/pull/4762/commits/bf75084518dc1ed3a1485670825356d8e444a931

Reply:
nit: it might be beneficial to emit these from inside the keeper. This will allow to integrate incentive collection into another part of the system if needed. So that if we call the keeper method without entering the message server flow here directly, the event is still emitted

Reply:
Added the event in the respective keepers here https://github.com/osmosis-labs/osmosis/pull/4762/commits/c6111c9eb3bb2c8b61b4d74ce092a2ef04ca9923